src/core/evaluator.js

```diff
@@ -2491,7 +2538,17 @@ class PartialEvaluator {
           appendEOL();
           return true;
         }
+
+        if (advanceY <= textOrientation * textContentItem.notASpace) {
+          // The real spacing between 2 consecutive chars is thin enough to be
+          // considered a non-space.
+          resetLastChars();
+        }
+
         if (advanceY <= textOrientation * textContentItem.trackingSpaceMin) {
+          if (advanceY <= textContentItem.notASpace) {
+            resetLastChars();
+          }
```

Comment:
There isn't a similar branch below for the `advanceX`-case down below, I'm assuming that's on purpose?

Reply:
It's a leftover: it has been replaced by the resetLastChars stuff just above.